src/api/java/jni/api_utilities.h

```diff
@@ -139,4 +139,22 @@ jobject getDoubleObject(JNIEnv* env, double value);
  */
 jobject getBooleanObject(JNIEnv* env, bool value);
 
+/**
+ * a map from solver pointers to global references that need to be freed when
+ * the java Solver.close method is called
+ */
+inline std::map<jlong, std::vector<jobject> > globalReferences;
```

Comment:
We really need to rethink memory management, but let's leave this to future PRs.

Reply:
This was my main review point for this PR. I guess we can handle this later.